eth/api.go

```diff
@@ -439,10 +448,42 @@ func formatError(err error) string {
 	return err.Error()
 }
 
+type timeoutError struct {}
+
+func (t *timeoutError) Error() string {
+	return "Execution time exceeded"
+}
+
 // TraceTransaction returns the structured logs created during the execution of EVM
 // and returns them as a JSON object.
-func (api *PrivateDebugAPI) TraceTransaction(txHash common.Hash, logConfig *vm.LogConfig) (*ethapi.ExecutionResult, error) {
-	logger := vm.NewStructLogger(logConfig)
+func (api *PrivateDebugAPI) TraceTransaction(ctx context.Context, txHash common.Hash, config *TraceArgs) (interface{}, error) {
+	var tracer vm.Tracer
+	if config != nil && config.Tracer != nil {
+		timeout := 5 * time.Second
```

Comment:
Nononononooo :D Please move this out into a constant to the top of the file :D :P